kubernetes-manifests/traefik-ingress-controller.yml

```diff
@@ -0,0 +1,117 @@
+---
+# this file contains configuration used to set up the traefik reverse proxy as an
+# ingress.
+# See https://kubernetes.io/docs/concepts/services-networking/ingress/
+# See https://docs.traefik.io/user-guide/kubernetes/
+
+# this is a bunch of roles that traefik needs in order to read information from the kubernetes API
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: traefik-ingress-controller
+rules:
+  - apiGroups:
+      - ""
+    resources:
+      - services
+      - endpoints
+      - secrets
+    verbs:
+      - get
+      - list
+      - watch
+  - apiGroups:
+      - extensions
+    resources:
+      - ingresses
+    verbs:
+      - get
+      - list
+      - watch
+---
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: traefik-ingress-controller
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: traefik-ingress-controller
+subjects:
+  - kind: ServiceAccount
+    name: traefik-ingress-controller
+    namespace: kube-system
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: traefik-ingress-controller
+  namespace: kube-system
+---
+# This ensures that there is exactly one traefik instance per running kubernetes node
+# This way, any of the IPs in the cluster can respond to any incoming requests
+# Note that this service uses the `hostPort` directly
+# See https://kubernetes.io/docs/concepts/workloads/controllers/daemonset/
+kind: DaemonSet
+apiVersion: extensions/v1beta1
+metadata:
+  name: traefik-ingress-controller
+  namespace: kube-system
+  labels:
+    k8s-app: traefik-ingress-lb
+spec:
+  template:
+    metadata:
+      labels:
+        k8s-app: traefik-ingress-lb
+        name: traefik-ingress-lb
+    spec:
+      tolerations:
+        # This tolerance is required, since Kubernetes no longer schedules DaemonSet instances
```

Comment:
According to https://twitter.com/renebakx/status/1070029749135835138, it is possible to configure kubernetes master to be used for scheduling work. Usually, master nodes are lightweight machines, but in our case it's raspberries through the entire stack.